TYP: incorrect array_finalize definition in ndarray for subclassing

https://github.com/numpy/numpy/blob/4adc87dff15a247e417d50f10cc4def8e1c17a03/numpy/__init__.pyi#L1505-L1506

Currently, mypy does not allow users to define `__array_finalize__`, since the super class (ndarray) does not have an argument. So the following error is produced:


```
src/vector/_backends/numpy_.py:899: error: Signature of "__array_finalize__" incompatible with supertype "ndarray"  [override]
src/vector/_backends/numpy_.py:899: note:      Superclass:
src/vector/_backends/numpy_.py:899: note:          def __array_finalize__(self) -> None
src/vector/_backends/numpy_.py:899: note:      Subclass:
src/vector/_backends/numpy_.py:899: note:          def __array_finalize__(self, obj: Any) -> None
```

This type stub should have a `obj: Any` argument.

--- a/numpy/__init__.pyi
+++ b/numpy/__init__.pyi
@@ -1502,8 +1502,7 @@ class ndarray(_ArrayOrScalarCommon, Generic[_ShapeType, _DType_co]):
         kwargs: Mapping[str, Any],
     ) -> Any: ...
 
-    @property
-    def __array_finalize__(self) -> None: ...
+    __array_finalize__: Any
 
     def __array_wrap__(
         self,

